rest/model/admin.go

```diff
@@ -6,38 +6,63 @@ import (
 	"github.com/pkg/errors"
 )
 
+func NewConfigModel() *APIAdminSettings {
+	return &APIAdminSettings{
+		Alerts:       &APIAlertsConfig{},
+		Amboy:        &APIAmboyConfig{},
+		Api:          &APIapiConfig{},
+		AuthConfig:   &APIAuthConfig{},
+		Credentials:  map[string]string{},
+		Expansions:   map[string]string{},
+		HostInit:     &APIHostInitConfig{},
+		Jira:         &APIJiraConfig{},
+		Keys:         map[string]string{},
+		LoggerConfig: &APILoggerConfig{},
+		NewRelic:     &APINewRelicConfig{},
+		Notify:       &APINotifyConfig{},
+		Plugins:      map[string]map[string]interface{}{},
+		Providers:    &APICloudProviders{},
+		RepoTracker:  &APIRepoTrackerConfig{},
+		Scheduler:    &APISchedulerConfig{},
+		ServiceFlags: &APIServiceFlags{},
+		Slack:        &APISlackConfig{},
+		Splunk:       &APISplunkConnectionInfo{},
+		Ui:           &APIUIConfig{},
+	}
+}
+
 // APIAdminSettings is the structure of a response to the admin route
 type APIAdminSettings struct {
-	Alerts             APIAlertsConfig                   `json:"alerts"`
-	Amboy              APIAmboyConfig                    `json:"amboy"`
-	Api                APIapiConfig                      `json:"api"`
-	ApiUrl             APIString                         `json:"api_url"`
-	AuthConfig         APIAuthConfig                     `json:"auth"`
-	Banner             APIString                         `json:"banner"`
-	BannerTheme        APIString                         `json:"banner_theme"`
-	ClientBinariesDir  APIString                         `json:"client_binaries_dir"`
-	ConfigDir          APIString                         `json:"configdir"`
-	Credentials        map[string]APIString              `json:"credentials"`
-	Expansions         map[string]APIString              `json:"expansions"`
-	GithubPRCreatorOrg APIString                         `json:"github_pr_creator_org"`
-	HostInit           APIHostInitConfig                 `json:"hostinit"`
-	IsNonProd          bool                              `json:"isnonprod"`
-	Jira               APIJiraConfig                     `json:"jira"`
-	Keys               map[string]APIString              `json:"keys"`
-	LoggerConfig       APILoggerConfig                   `json:"logger_config"`
-	LogPath            APIString                         `json:"log_path"`
-	NewRelic           APINewRelicConfig                 `json:"new_relic"`
-	Notify             APINotifyConfig                   `json:"notify"`
+	Alerts             *APIAlertsConfig                  `json:"alerts"`
+	Amboy              *APIAmboyConfig                   `json:"amboy"`
+	Api                *APIapiConfig                     `json:"api"`
+	ApiUrl             *string                           `json:"api_url"`
+	AuthConfig         *APIAuthConfig                    `json:"auth"`
+	Banner             *string                           `json:"banner"`
```

Comment:
Don't these still need to be APIString?

Reply:
I don't believe so - the only difference between APIstring and string is that "null" will unmarshal to a pointer to a nil value, which I don't need with this change